### PR TITLE
Useful questions for Data Reference

### LINKED ISSUE #3153: edit bar and save / cancel not showing up after starting to edit an entity in the data reference
Looks like when you hit edit on segments or metrics the bar is taking up space but not visible which makes it impossible to save edits.
screen shot 2016-08-09 at 12 03 54 pm

*  I am using Chrome v52
*  My computer's OS is Mac OSX
*  My data is in a H2 database. 
*  My Metabase version is HEAD.

### PR DESCRIPTION
**Pull request by lewisl9029**
Heroku: https://metabase-useful-questions.herokuapp.com/reference/guide

Resolves #3153 

TODOs:
- [x] Switch existing links to query builder to use serialized card URLs
- [x] Add useful questions for Segments
- [x] Add useful questions for Tables
- [x] Add useful questions for Fields
- [x] Add alternative visualizations for Field "Number of x grouped by y" questions
- [x] Add "Subset of x" link to Segment headers
- [x] Add formula viewer for Metrics
- [x] Add formula viewer for Segments
- [x] Add fields to group by for Metrics
- [x] Fix buggy animation for formula viewer when expanding and collapsing
- [x] Add tests
- [x] Refactor everything into components

### CODE REVIEW

File: frontend/src/metabase/components/Detail.css [hunk 1]
```diff
@@ -3,12 +3,13 @@
     --subtitle-color: #AAB7C3;
     --muted-color: #DEEAF1;
     --blue-color: #2D86D4;
+    --icon-width: calc(48px + 1rem);
```
Comment (tlrobinson) [inline frontend/src/metabase/components/Detail.css]: @kdoh Is `calc` kosher?

Comment (lewisl9029) [inline frontend/src/metabase/components/Detail.css]: The alternative I had before was combining a pixel based padding and rem based margin to achieve the same layout, which feels much worse in my opinion, both in terms of readability and flexibility for future layout changes. Definitely open to hearing about other possible alternatives though.

File: frontend/src/metabase/redux/metadata.js [hunk 2]
```diff
@@ -333,10 +333,20 @@ const revisions = handleActions({
...
+        const tableId = metric.table_id;
+        await dispatch(fetchTableMetadata(tableId));
+    };
+});
+
 const FETCH_METRIC_REVISIONS = "metabase/metadata/FETCH_METRIC_REVISIONS";
 export const fetchMetricRevisions = createThunkAction(FETCH_METRIC_REVISIONS, (metricId, reload = false) => {
     return async (dispatch, getState) => {
-        dispatch(fetchRevisions('metric', metricId));
+        await dispatch(fetchRevisions('metric', metricId));
```
Comment (tlrobinson) [inline frontend/src/metabase/redux/metadata.js]: Might want to do these in parallel, e.x.

```
await Promise.all([
  dispatch(fetchRevisions('metric', metricId)),
  dispatch(fetchMetrics())
]);
```

Comment (lewisl9029) [inline frontend/src/metabase/redux/metadata.js]: Good idea. Done. =)

**Comment by tlrobinson** [comment]
lgtm

**Pull request merged**